File: packages/core/src/services/noop-service.ts
import type { FuzeService, ToolRegistration, ToolConfig, StepCheckData, StepEndData, GuardEventData } from './types.js'

/**
 * No-op service — used when neither daemon nor cloud is configured.
 * All operations succeed instantly; step checks always return 'proceed';
 * tool config is never fetched (SDK uses local defaults).
 */
export class NoopService implements FuzeService {
  async connect(): Promise<boolean> { return true }
  async disconnect(): Promise<void> {}
  isConnected(): boolean { return true }
  async flush(): Promise<void> {}

  async registerTools(_projectId: string, _tools: ToolRegistration[]): Promise<void> {}
  getToolConfig(_toolName: string): ToolConfig | null { return null }
  async refreshConfig(): Promise<void> {}

  async sendRunStart(_runId: string, _agentId: string, _config: object): Promise<void> {}
  async sendStepStart(_runId: string, _step: StepCheckData): Promise<'proceed' | 'kill' | 'pause'> {
    return 'proceed'
  }
  async sendStepEnd(_runId: string, _stepId: string, _data: StepEndData): Promise<void> {}
  async sendGuardEvent(_runId: string, _event: GuardEventData): Promise<void> {}
  async sendRunEnd(_runId: string, _status: string): Promise<void> {}
}

File: packages/python/src/fuze_ai/services/noop_service.py
from __future__ import annotations

from typing import Any, Literal

from .types import FuzeService, GuardEventData, StepCheckData, StepEndData, ToolConfig, ToolRegistration


class NoopService(FuzeService):
    async def connect(self) -> bool:
        return True

    async def disconnect(self) -> None:
        return None

    def is_connected(self) -> bool:
        return True

    async def flush(self) -> None:
        return None

    async def register_tools(self, project_id: str, tools: list[ToolRegistration]) -> None:
        return None

    def get_tool_config(self, tool_name: str) -> ToolConfig | None:
        return None

    async def refresh_config(self, force: bool = False) -> None:
        return None

    async def send_run_start(self, run_id: str, agent_id: str, config: dict[str, Any]) -> None:
        return None

    async def send_step_start(self, run_id: str, step: StepCheckData) -> Literal["proceed", "kill", "pause"]:
        return "proceed"

    async def send_step_end(self, run_id: str, step_id: str, data: StepEndData) -> None:
        return None

    async def send_guard_event(self, run_id: str, event: GuardEventData) -> None:
        return None

    async def send_run_end(self, run_id: str, status: str) -> None:
        return None
